sphinx/util/typing.py

```diff
@@ -365,7 +365,8 @@ def format_literal_arg(arg):
             args = ', '.join(map(format_literal_arg, annotation_args))
             return f'{module_prefix}Literal[{args}]'
         elif str(annotation).startswith('typing.Annotated'):  # for py39+
-            return stringify_annotation(annotation_args[0], mode)
+            meta = [stringify_annotation(meta, mode) for meta in annotation.__metadata__]
```

Comment:
Metadata is actually arbitrary so I'm not sure what should be done here. Here, you assume that your metadata are annotations but they can be non-annotations (like real values). So what should we do? 

Actually, I want to delay whatever rendering issues we have currently because we might change the logic of how rendering is done, e.g., by allowing custom rendering for each type of components.